Lib/DHMap.hpp

```diff
@@ -323,7 +335,7 @@ class DHMap
    * the value with @b initial, and return true. Otherwise,
    * return false.
    */
-  bool getValuePtr(Key key, Val*& pval, const Val& initial)
+  bool getValuePtr(Key key, Val*& pval, Val initial)
   {
```

Comment:
Did you somehow check this does not affect other users of this function?

(Shouldn't matter for small `Val`, but could for large...)

Reply:
I see your point... How about adding one more variant with rvalue reference `Val&&`? (`DHMap` could also use some refactor as it contains lots of duplication, although maybe not the scope of this PR.)

Alternatively, I can just try to use a different structure, or try really hard to make it work with the current interface.